torch/distributed/_tensor/experimental/tensor_parallel.py

```diff
@@ -0,0 +1,426 @@
+import copy
```

Comment:
nit: let's also rename this file to sth like `tp_transform.py` or `tensor_parallel_transform.py` to make this more explicit that this is a graph transformation pass.